examples/greeter-extension/src/types.ts

```diff
@@ -0,0 +1,33 @@
+// Copyright IBM Corp. 2019. All Rights Reserved.
+// Node module: @loopback/example-greeter-extension
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {BindingScope, BindingTemplate} from '@loopback/context';
+
+/**
+ * Typically an extension point defines an interface as the contract for
+ * extensions to implement
+ */
+export interface Greeter {
+  language: string;
+  greet(name: string): string;
+}
+
+/**
+ * A factory function to create binding template for extensions of the given
+ * extension point
+ * @param extensionPoint Name/id of the extension point
+ */
+export function extensionFor(extensionPoint: string): BindingTemplate {
+  return binding =>
+    binding.inScope(BindingScope.SINGLETON).tag({extensionPoint});
+}
+
+/**
+ * A binding template for greeter extensions
+ */
+export const asGreeter: BindingTemplate = binding => {
+  extensionFor('greeter')(binding);
```

Comment:
Something maybe worth mentioning here:
The extension name here should be consistent with the one in the [extensionPoint decorator](https://github.com/strongloop/loopback-next/blob/988dbb62511c61183b7ad3c3b305b8007fee52b7/examples/greeter-extension/src/greeting-service.ts#L21)